tests/libgstc/c/test_libgstc_pipeline_get_graph.c

```diff
@@ -0,0 +1,174 @@
+{
+  *out = &_socket;
+
+  return GSTC_OK;
+}
+
+void
+gstc_socket_free (GstcSocket * socket)
+{
+}
+
+GstcStatus
+gstc_socket_send (GstcSocket * socket, const gchar * request, gchar ** response,
+    const int timeout)
+{
+  *response = malloc (1);
+
+  memcpy (_request, request, strlen (request));
+
+  return GSTC_OK;
+}
+
+GstcStatus
+gstc_json_get_int (const gchar * json, const gchar * name, gint * out)
+{
+  return *out = GSTC_OK;
+}
+
+GstcStatus
+gstc_json_is_null (const gchar * json, const gchar * name, gint * out)
+{
+  *out = 0;
+  return GSTC_OK;
+}
+
+GstcStatus
+gstc_json_get_child_char_array (const char *json, const char *parent_name,
+    const char *array_name, const char *element_name, char **out[],
+    int *array_lenght)
+{
+  gstc_assert_and_ret_val (NULL != json, GSTC_NULL_ARGUMENT);
+  gstc_assert_and_ret_val (NULL != parent_name, GSTC_NULL_ARGUMENT);
+  gstc_assert_and_ret_val (NULL != array_name, GSTC_NULL_ARGUMENT);
+  gstc_assert_and_ret_val (NULL != element_name, GSTC_NULL_ARGUMENT);
+  gstc_assert_and_ret_val (NULL != out, GSTC_NULL_ARGUMENT);
+  gstc_assert_and_ret_val (NULL != array_lenght, GSTC_NULL_ARGUMENT);
+  return GSTC_OK;
+}
+
+GstcStatus
+gstc_json_child_string (const char *json, const char *parent_name,
+    const char *data_name, char **out)
+{
+  gstc_assert_and_ret_val (NULL != json, GSTC_NULL_ARGUMENT);
+  gstc_assert_and_ret_val (NULL != parent_name, GSTC_NULL_ARGUMENT);
+  gstc_assert_and_ret_val (NULL != data_name, GSTC_NULL_ARGUMENT);
+  gstc_assert_and_ret_val (NULL != out, GSTC_NULL_ARGUMENT);
+
+  return GSTC_OK;
+}
+
+GST_START_TEST (test_pipeline_get_graph_success)
+{
+  GstcStatus ret;
+  const gchar *pipeline_name = "pipe";
+  const gchar *expected = "read /pipelines/pipe/graph";
+  char *pipe_description = NULL;
+
+  ret = gstc_pipeline_get_graph (_client, pipeline_name, &pipe_description);
+  assert_equals_int (GSTC_OK, ret);
+
+  assert_equals_string (expected, _request);
+}
+
+GST_END_TEST;
+
+GST_START_TEST (test_pipeline_get_graph_null_name)
+{
+  GstcStatus ret;
+  const gchar *pipeline_name = NULL;
+  char *pipe_description = NULL;
+
+  ret = gstc_pipeline_get_graph (_client, pipeline_name, &pipe_description);
+  assert_equals_int (GSTC_NULL_ARGUMENT, ret);
+}
+
+GST_END_TEST;
+
+GST_START_TEST (test_pipeline_get_graph_null_client)
+{
+  GstcStatus ret;
+  const gchar *pipeline_name = "pipe";
+  char *pipe_description = NULL;
+
+  ret = gstc_pipeline_get_graph (NULL, pipeline_name, &pipe_description);
+  assert_equals_int (GSTC_NULL_ARGUMENT, ret);
+}
+
+GST_END_TEST;
+
+static Suite *
+libgstc_pipeline_suite (void)
+{
+  Suite *suite = suite_create ("libgstc_pipeline_get_graph");
+  TCase *tc = tcase_create ("general");
+
+  suite_add_tcase (suite, tc);
+
+  tcase_add_checked_fixture (tc, setup, teardown);
+  tcase_add_test (tc, test_pipeline_get_graph_success);
+  tcase_add_test (tc, test_pipeline_get_graph_null_name);
+  tcase_add_test (tc, test_pipeline_get_graph_null_client);
```

Comment:
What about null placeholder for output?

Reply:
Done with: test_pipeline_get_graph_null_output